python/pyiceberg/expressions/literals.py

```diff
@@ -213,6 +217,26 @@ class LongLiteral(Literal[int]):
     def __init__(self, value: int):
         super().__init__(value, int)
 
+    def __add__(self, other: Any) -> LongLiteral:
```

Comment:
What about `increment` and `decrement` methods since we only need to add or subtract 1?

I think I like the idea of implementing this on the literal. The main issue that I have with this is that it modifies the value of the literal. I think introducing mutability is going to cause problems because this will modify the original expression when we project, which should not happen. (That may also be why expressions are no longer immutable?)

I think this should just return a new literal rather than `self`.

Reply:
Good call on the mutability. Python doesn't have a `++` or `--`, this is why I liked overriding `__add__` and `__sub__`. Changed this into `.increment()` and `.decrement()`.